File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import math

limit = int(123456) * 2
prime = [True] * (limit + 1)
prime[1] = False
for i in range(2, int(math.sqrt(limit)) + 1):
    if (prime[i] == True):
        j = 2
        while i * j <= limit:
            prime[i * j] = False
            j += 1
            
            
while(1):
    n = int(input())
    if(n == 0): break
    hap = 0
    for i in range(n + 1, 2 * n + 1):
        if(prime[i]):
            hap += 1
    print(hap)

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define LIMIT (123456 * 2)

int main(void) {
	int* prime = (int*)malloc(sizeof(int) * (LIMIT + 1));
	memset(prime, 1, sizeof(int) * (LIMIT + 1));
	prime[1] = 0;
	for (int i = 2; i < LIMIT + 1; i++) {
		int j = 2;
		while (i * j <= LIMIT) {
			prime[i * j] = 0;
			j++;
		}
	}
	while (1) {
		int n, hap = 0;
		scanf("%d", &n);
		if (n == 0) break;
		for (int i = n + 1; i < 2 * n + 1; i++) {
			if (prime[i]) hap++;
		}
		printf("%d\n", hap);
	}
	
	return 0;
}